pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -749,17 +765,14 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	// *** At this point the volume population is done and we're just cleaning up ***
 	c.recorder.Eventf(pvc, corev1.EventTypeNormal, reasonPodFinished, "Populator finished")
 
-	// If the pod still exists, delete it
-	if pod != nil {
-		err = c.kubeClient.CoreV1().Pods(populatorNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+	// If PVC' still exists, delete it
+	if pvcPrime != nil {
+		err = c.kubeClient.CoreV1().PersistentVolumeClaims(populatorNamespace).Delete(ctx, pvcPrime.Name, metav1.DeleteOptions{})
 		if err != nil {
 			return err
 		}
-	}
 
-	// If PVC' still exists, delete it
-	if pvcPrime != nil {
-		err = c.kubeClient.CoreV1().PersistentVolumeClaims(populatorNamespace).Delete(ctx, pvcPrime.Name, metav1.DeleteOptions{})
+		err = c.ensureFinalizer(ctx, pvcPrime, pvcKuberenetesFinalizer, false)
```

Comment:
I'm not sure about this, why is this needed? This is the kubernetes finalizer and it could lead to CSI not cleaning up the underlying storage

Reply:
it's the other way around, it unsure to remove the finalizer. without it we will get stuck with the prime pvc.

Reply:
Stuck because of what though? Since we are not the ones who set `kubernetes.io/pvc-protection` in theory we are not supposed to remove it

Reply:
maybe it's stuck because of the pvcinit pod?

Reply:
yeah I wrote it when I thought we don't keep the populator after successful import...